step.sh

```diff
@@ -1,82 +1,124 @@
 #!/usr/bin/env bash
-# fail if any commands fails
+# fail if any commands fail
 set -e
 # debug log
 if [ "${show_debug_logs}" == "yes" ]; then
   set -x
 fi
 
-function getToken()
-{
-  printf "\n\nObtaining a Token\n"  
-  
-  curl --silent -X POST \
+function getToken() {
+  printf "\n\nObtaining a Token...\n"
+
+  response=$(curl --silent -X POST \
     https://connect-api.cloud.huawei.com/api/oauth2/v1/token \
     -H 'Content-Type: application/json' \
     -H 'cache-control: no-cache' \
     -d '{
       "grant_type": "client_credentials",
       "client_id": "'${huawei_client_id}'",
       "client_secret": "'${huawei_client_secret}'"
-  }' > token.json
+  }' || true)
+
+  if [[ -z "$response" ]]; then
+    printf "\n ❌ Failed to obtain a token. Check your network connection and credentials 😢\n"
+    exit 1
+  fi
+
+  echo "$response" >token.json
 
-  printf "\nObtaining a Token ✅\n"
-} 
+  CODE=$(jq -r '.ret.code' token.json)
+  if [ "${CODE}" != "null" ] && [ "${CODE}" != "0" ]; then
+    printf "\n ❌ Failed to obtain a token 😢\n"
```

Comment:
[nit] Maybe this would have to be a bit more descriptive, so the user will be able to identify that the cause of the error came from the code. @jakob-grabner

Reply:
I tried to improve the message, the cause of the error should be part of the response that gets printed after this message.